spec/components/post_merger_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+require 'post_merger'
+
+describe PostMerger do
+
+  before do
+    ActiveRecord::Base.observers.enable :all
+  end
+
+  let(:moderator) { Fabricate(:moderator) }
+  let(:admin) { Fabricate(:admin) }
+  let(:post) { create_post }
+
+  describe "merge posts" do
+
+    it "merges 3 posts" do
+      Fabricate(:admin)
+      topic = post.topic
+      reply1 = create_post(topic: topic)
+      reply2 = create_post(topic: topic)
+      reply3 = create_post(topic: topic)
+
+      replies = []
+      replies.push(reply1)
+      replies.push(reply2)
+      replies.push(reply3)
+
+      postContent = []
+      replies.each {|p| postContent.push(p.raw) }
```

Comment:
Hmm I'm not a fan of this test because it is duplicating the logic of what `PostMerger` does in the test. 

What I usually do is manually form the expected output, so it'll be something like

```
create_post(topic: topic, raw: 'this is the first line')
create_post(topic: topic, raw: 'this is the second line')
create_post(topic: topic, raw: 'this is the third line')

expected_output = <<~OUTPUT
this is the first line

this is the second line

this is the third line
OUTPUT
```